File: _java/symulacja_organizmow/src/Symulacja/Rosliny/Trawa.java
package Symulacja.Rosliny;
import Symulacja.*;

import java.awt.*;

public class Trawa extends Roslina {
    public Trawa(Swiat swiat){
        super(swiat);
    };
    public char Rysowanie(){
        return 't';
    };
    public Organizm Urodz(){
        return new Trawa(this.swiat);
    };
    public int GetPrawdopodobienstwo(){
        return 30;
    };
    public boolean MozeBycPartnerem(Organizm organizm){
        if(organizm instanceof Trawa)
           return true;
        else
          return false;
    };
    @Override
    public Color GetColor(){
        return new Color(255,255,0);
    };
}

File: _c++/symulacja_organizmow/Organizmy/Rosliny/Trawa.cpp
//
//  Trawa.cpp
//  WirtualnySwiat
//
//  Created by Michal Tarnacki on 10/04/2022.
//

#include "Trawa.hpp"

Trawa::Trawa(Swiat &swiat):Roslina(swiat){
}

char Trawa::Rysowanie() const{
    return 't';
}
Organizm* Trawa::Urodz(){
    return new Trawa(this->swiat);
}
int Trawa::GetPrawdopodobienstwo(){
    return 30;
}

bool Trawa::MozeBycPartnerem(Organizm& organizm){
    if(dynamic_cast<Trawa*>(&organizm))
       return true;
    else
       return false;
}
